### PR TITLE
Detect supported format in rest api action to choose between cbor/json

### PR DESCRIPTION
**Pull request by lucblender**

### CODE REVIEW

File: src/main/kotlin/ch/hevs/cloudio/cloud/apiutils/EndpointManagementUtil.kt [hunk 1]
```diff
@@ -116,10 +120,22 @@ object EndpointManagementUtil {
         else {
             //send message to amq.topic queue
 
-            // TODO: Detect actual serialization format from endpoint data model.
+            val endpointUUID = attributeSetRequest.attributeTopic.split("/")[0]
+            val endpoint = endpointEntityRepository.findById(UUID.fromString(endpointUUID)).get()
+            val serializedAttribute =
+                if(endpoint.endpoint.supportedFormats.contains("JSON")){
+                    println("JSON")
```
Comment (cm0x4D) [inline src/main/kotlin/ch/hevs/cloudio/cloud/apiutils/EndpointManagementUtil.kt]: The evil println adder has hit again ;-)

File: src/main/kotlin/ch/hevs/cloudio/cloud/apiutils/EndpointManagementUtil.kt [hunk 2]
```diff
@@ -116,10 +120,22 @@ object EndpointManagementUtil {
         else {
             //send message to amq.topic queue
 
-            // TODO: Detect actual serialization format from endpoint data model.
+            val endpointUUID = attributeSetRequest.attributeTopic.split("/")[0]
+            val endpoint = endpointEntityRepository.findById(UUID.fromString(endpointUUID)).get()
+            val serializedAttribute =
```
Comment (cm0x4D) [inline src/main/kotlin/ch/hevs/cloudio/cloud/apiutils/EndpointManagementUtil.kt]: Check out the solution I did in the PostgreSQL branch:
```kotlin
fun Collection<SerializationFormat>.fromIdentifiers(identifiers: Collection<String>) = this.firstOrNull {
    (if (identifiers.isEmpty()) listOf("JSON") else identifiers).contains(it.identifier())
}
```

Then we can just:

```kotlin
val serializationFormat = serializationFormats.fromIdentifiers(endpoint.dataModel.supportedFormats)
                        ?: throw CloudioHttpExceptions.InternalServerError("Endpoint does not support any serialization format.")
                rabbitTemplate.convertAndSend("amq.topic",
                        modelIdentifier.toAmqpTopic(),
                        serializationFormat.serializeAttribute(Attribute(
                                data.constraint,
                                data.type,
                                System.currentTimeMillis().toDouble() / 1000,
                                typedValue)))
```

**Review by cm0x4D — CHANGES_REQUESTED** [review]

File: src/main/kotlin/ch/hevs/cloudio/cloud/apiutils/EndpointManagementUtil.kt (hunk 1 above)
Comment (lucblender) [reply src/main/kotlin/ch/hevs/cloudio/cloud/apiutils/EndpointManagementUtil.kt]: It keeps trying to enter my code, it's an everyday battle x)

File: src/main/kotlin/ch/hevs/cloudio/cloud/apiutils/EndpointManagementUtil.kt (hunk 2 above)
Comment (lucblender) [reply src/main/kotlin/ch/hevs/cloudio/cloud/apiutils/EndpointManagementUtil.kt]: Oh nice, I'll do it like that

**Review by cm0x4D — APPROVED** [review]

**Pull request merged**